pkg/controller/batchrelease/workloads/statefulset_like_controller.go

```diff
@@ -198,49 +191,40 @@ func (c *StatefulSetLikeController) IsBatchReady(canaryReplicasGoal, stableRepli
 	return secondCheckPointReady(), nil
 }
 
-func (c *StatefulSetLikeController) listOwnedPods() ([]*v1.Pod, error) {
+func (c *StatefulSetLikeController) ListOwnedPods() ([]*v1.Pod, error) {
 	if c.pods != nil {
 		return c.pods, nil
 	}
 	set, err := c.GetWorkloadObject()
 	if err != nil {
 		return nil, err
 	}
-	selector, err := util.ParseSelectorFrom(set)
-	if err != nil || selector == nil {
-		return nil, err
-	}
-	podLister := &v1.PodList{}
-	err = c.List(context.TODO(), podLister, &client.ListOptions{LabelSelector: selector}, utilclient.DisableDeepCopy)
-	if err != nil {
-		return nil, err
-	}
-	c.pods = make([]*v1.Pod, 0)
-	for i := range podLister.Items {
-		pod := &podLister.Items[i]
-		if !pod.DeletionTimestamp.IsZero() {
-			continue
-		}
-		owner := metav1.GetControllerOf(pod)
-		if owner == nil || owner.UID != set.GetUID() {
-			continue
-		}
-		c.pods = append(c.pods, pod)
-	}
-	return c.pods, nil
+	c.pods, err = util.ListOwnedPods(c.Client, set)
+	return c.pods, err
 }
 
 func (c *StatefulSetLikeController) countUpdatedReadyPods(updateRevision string) (int32, error) {
-	pods, err := c.listOwnedPods()
+	pods, err := c.ListOwnedPods()
 	if err != nil {
 		return 0, err
 	}
+	activePods := util.FilterActivePods(pods)
```

Comment:
can we reuse FilterActivePods in the controller_utils.go

Reply:
It may  introduce too many dependencies.